testutils/inject/analog.go

```diff
@@ -30,3 +32,21 @@ func (a *Analog) ReadCap() []interface{} {
 	defer func() { a.readCap = nil }()
 	return a.readCap
 }
+
+// Write calls the injected Write or the real version.
+func (a *Analog) Write(ctx context.Context, value int, extra map[string]interface{}) error {
+	a.readCap = []interface{}{ctx}
+	if a.ReadFunc == nil {
```

Comment:
Copying and pasting code is the bane of my existence and leads to lots of bugs. Both lines 38 and 39 should be about writing, not reading.

Reply:
That's why I ask for reviews.